packages/compass-web/src/entrypoint.tsx

```diff
@@ -277,15 +279,22 @@ const CompassWeb = ({
   const autoConnectConnectionId = initialWorkspaceRef.current?.connectionId;
 
   const telemetry = useRef({
-    createTrack:
-      () =>
-      async (event: string, properties: Record<string, any> | undefined) => {
-        onTrack &&
-          void onTrack(
-            event,
-            typeof properties === 'function' ? await properties() : properties
-          );
-      },
+    createTrack: (logger: Logger, preferences: TelemetryPreferences) =>
+      createGenericTrack({
+        sendTrack: async (
+          event: string,
+          properties: Record<string, any> | undefined
+        ) => {
+          onTrack &&
```

Comment:
Needs to be a ref too and not just directly used here from props, otherwise (because you are only defining this once) this is always a stale reference from the initial render and is not accounting for potential prop updates